packages/vendure-plugin-invoices/src/services/invoice.service.ts

```diff
@@ -264,9 +265,24 @@ export class InvoiceService implements OnModuleInit, OnApplicationBootstrap {
   async previewInvoiceWithTemplate(
     ctx: RequestContext,
     template: string,
-    orderCode: string
+    orderCode?: string
   ): Promise<ReadStream> {
-    const order = await this.orderService.findOneByCode(ctx, orderCode);
+    let order: Order | undefined;
+    if (orderCode) {
+      order = await this.orderService.findOneByCode(ctx, orderCode);
+    } else {
+      const orderId = (
+        await this.orderService.findAll(
+          ctx,
+          {
+            take: 1,
+            sort: { createdAt: SortOrder.DESC },
+          },
+          []
+        )
+      )?.items[0].id;
+      order = await this.orderService.findOne(ctx, orderId);
```

Comment:
1. Can we just get the latest placed order?
2. Why not use `items[0]` instead of using it's ID to refetch an order?

Reply:
`OrderService.findOne` includes more relations, translates entities and also applies prices to the `ProductVariant`'s in the Order. You don't think this is a good approach @martijnvdbrug ?